### ISSUE TITLE
Will you - and when if yes - come back to WP core development?

### ISSUE THREAD

**Issue by slaFFik** [Rarst/AMA#2]
As an example: Alain Schlesser is pushing nice things, do you want to help him to act faster and further?

**Comment by Rarst** [comment]
Well, there is no "back" since I never participated. :)

I believe I have long been quite open about avoiding core dev, but don't think I had a chance to put it in writing.

So if I had to enumerate why I don't participate in core development:

1. **Trac as a software**. I don’t want to dump on trac. Clearly it had served many people (not just WordPress project) well over many years. But I don’t find it enjoyable to use. It gets in my way and it’s not the way in which I do open source development anymore.

2. **WP trac as a culture**. I consider it to be significantly toxic social environment. I could deal with _just_ that, but I think over years that toxicity had been pretty much _weaponized_ to smother any "unwelcome" opinions and initiatives. I have quite a few of those.

3. **I only write PHP 5.2 for money**. I know this one sounds silly, but it had served well both my growth as a developer and earning a living. While I had offers to be paid for WP core development in the past, so far I don’t find them appealing for all the other reasons on the list.

4. **Misalignment of philosophy**. WordPress puts popularity above all. While popularity _is good_ for it (and me!), after certain point it doesn't matter _to me_, it's "enough". But WP remains insatiable for popularity and it had committed to unbelievably costly trade–offs to pay for it. For me WP matters as a _tool_ and while it progressed in many aspects, it failed in many others.

5. **Misalignment of technical vision**. I care about a lot of things on technical level that core development doesn’t. I find the rift it made with PHP development and the state of many (important!) parts of WP core are stuck in... embarrassing.

So if I had to summarize what could hypothetically interest me to get involved in core dev, I would roughly say WP project would have to **have an honest outlook on its state of Developer Experience and come up with a clear and actionable roadmap to resolve outstanding problems with it**.

And to answer your second question literally — Alain is doing great things in core dev and they are lucky to have him, but I think he will be just fine without me and my baggage. :)
